ecs-cli/modules/cli/local/localproject/project.go

```diff
@@ -200,68 +202,91 @@ var readTaskDefFromRemote = func(remote string, p *localProject) (*ecs.TaskDefin
 
 // Convert translates an ECS Task Definition into a Compose V3 schema and
 // stores the data on the project
-func (p *localProject) Convert() error {
+func (p *LocalProject) Convert() error {
+	if err := p.convertBaseCompose(); err != nil {
+		return err
+	}
+	if err := p.convertOverrideCompose(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *LocalProject) convertBaseCompose() error {
 	conf, err := converter.ConvertToComposeConfig(p.taskDefinition, p.inputMetadata)
 	if err != nil {
 		return err
 	}
-	data, err := converter.MarshalComposeConfig(*conf, LocalOutDefaultFileName)
+	data, err := converter.MarshalComposeConfig(*conf, p.LocalOutFileName())
 	if err != nil {
 		return err
 	}
+	p.baseCompose = data
+	return nil
+}
 
-	p.localBytes = data
+func (p *LocalProject) convertOverrideCompose() error {
+	conf, err := converter.ConvertToComposeOverride(p.taskDefinition)
+	if err != nil {
+		return err
+	}
+	data, err := converter.MarshalComposeConfig(*conf, p.OverrideFileName())
+	if err != nil {
+		return err
+	}
+	p.overrideCompose = data
 	return nil
 }
 
 // Write writes the compose data to a local compose file. The output filename is stored on the project
-func (p *localProject) Write() error {
-	// Will error if the file already exists, otherwise create
-	p.localOutFileName = LocalOutDefaultFileName
+func (p *LocalProject) Write() error {
+	if err := p.writeFile(p.LocalOutFileName(), p.baseCompose); err != nil {
```

Comment:
This all looks fine, but I do wonder if we are now being a little inconsistent with methods and data attached to the project struct versus passing things in as arguments (e.g. I wouldn't expect a method, i.e. `p.writeFile` to take an argument that is a member of the thing itself, i.e. `p.baseCompose`) but I could be convinced otherwise.

Reply:
🤦‍♂️ agreed, I actually didn't realize that `writeFile` and `overrideFile` were methods.
Changed them to functions so that they can be re-used for the override file.